File: Core/YABACore/src/commonMain/kotlin/dev/subfly/yabacore/state/detail/tag/TagDetailEvent.kt
package dev.subfly.yabacore.state.detail.tag

import dev.subfly.yabacore.model.ui.BookmarkUiModel
import dev.subfly.yabacore.model.utils.BookmarkAppearance
import dev.subfly.yabacore.model.utils.CardImageSizing
import dev.subfly.yabacore.model.utils.SortOrderType
import dev.subfly.yabacore.model.utils.SortType
import kotlin.uuid.ExperimentalUuidApi
import kotlin.uuid.Uuid

@OptIn(ExperimentalUuidApi::class)
sealed interface TagDetailEvent {
    data class OnInit(val tagId: String) : TagDetailEvent
    data class OnChangeQuery(val query: String) : TagDetailEvent
    data object OnToggleSelectionMode : TagDetailEvent
    data class OnToggleBookmarkSelection(val bookmarkId: Uuid) : TagDetailEvent
    data object OnDeleteSelected : TagDetailEvent
    data class OnDeleteBookmark(val bookmark: BookmarkUiModel) : TagDetailEvent
    data class OnChangeSort(val sortType: SortType, val sortOrder: SortOrderType) : TagDetailEvent
    data class OnChangeAppearance(
        val appearance: BookmarkAppearance,
        val cardImageSizing: CardImageSizing? = null
    ) : TagDetailEvent
}

File: Darwin/YABACore/State/Detail/Tag/TagDetailEvent.swift
//
//  TagDetailEvent.swift
//  YABACore
//
//  Parity with Compose `TagDetailEvent` + tag delete/hide.
//

import Foundation

public enum TagDetailEvent: Sendable {
    case onInit(tagId: String)
    case onChangeQuery(String)
    case onToggleSelectionMode
    case onToggleBookmarkSelection(bookmarkId: String)
    case onDeleteSelected(bookmarkIds: [String])
    case onDeleteBookmark(bookmarkId: String)
    case onChangeSort(sortType: SortType, sortOrder: SortOrderType)
    case onChangeAppearance(appearance: BookmarkAppearance, cardImageSizing: CardImageSizing?)

    case deleteOrHide(tagId: String)
}
